be/src/exec/parquet_scanner.cpp

```diff
@@ -98,16 +105,29 @@ Status ParquetScanner::append_batch_to_src_chunk(ChunkPtr* chunk) {
     SCOPED_RAW_TIMER(&_counter->fill_ns);
     size_t num_elements =
             std::min<size_t>((_max_chunk_size - _chunk_start_idx), (_batch->num_rows() - _batch_start_idx));
-    size_t column_pos = 0;
     _chunk_filter.resize(_chunk_filter.size() + num_elements, 1);
     for (auto i = 0; i < _num_of_columns_from_file; ++i) {
         SlotDescriptor* slot_desc = _src_slot_descriptors[i];
         if (slot_desc == nullptr) {
             continue;
         }
         _conv_ctx.current_slot = slot_desc;
-        auto* array = _batch->column(column_pos++).get();
+        auto* array = _batch->GetColumnByName(slot_desc->col_name()).get();
         auto& column = (*chunk)->get_column_by_slot_id(slot_desc->id());
+
+        // The column name is not found in the parquet file
+        if (array == nullptr) {
+            // append null.
+            column->append_default(num_elements);
```

Comment:
append default or null?

Reply:
The implement of `Nullable:append_default()` is `append_nulls()`.